src/main/kotlin/com/faforever/userservice/domain/UserService.kt

```diff
@@ -112,6 +119,16 @@ class UserService(
                         internalLogin(challenge, usernameOrEmail, password, ip, loginRequest)
                     }
             }
+        }.onErrorResume(HttpClientResponseException::class) {
+            if (it.status == HttpStatus.GONE) {
+                it.response.getBody(RequestWasHandledResponse::class.java)
+                    .map { LoginResult.SuccessfulLogin(it.redirectTo) }
+                    .orElseThrow()
+                    .toMono()
+            } else {
+                // pass through unknown error
+                Mono.error(it)
+            }
```

Comment:
If this could be a function that takes a lambda for the map that would be nice but is just a nit